### PR TITLE
[WIP] agate: 1.2.2 -> 1.5.5

### PR DESCRIPTION
**Pull request by vrthra**
###### Motivation for this change

Updating the Agate python library, and its dependencies. All these packages are connected, and I need to enable Python3 for these too. So adding them as one single PR

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by vrthra** [comment]
The error seems to be `no space left on the device`, and an unrelated error on `python cairocffi`. They don't seem related to my commits

**Comment by vrthra** [comment]
Moving to WIP: I am trying to split this PR into multiple smaller PRs as much as possible.

**Comment by FRidh** [comment]
@vrthra if these packages/updates are all needed for your agate update, then do not split it into multiple PR's.

**Comment by FRidh** [comment]
also, please put new expressions in `pkgs/development/python-modules/`

**Comment by vrthra** [comment]
@FRidh thanks, I didn't know if my PR was too large for review. I will add them back, and use `python-modules` for new expressions.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -144,14 +160,36 @@ in {
       maintainers = with maintainers; [ vrthra ];
     };
 
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: please include a comments explaining why you disabled the tests

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by vrthra** [comment]
Moving to #25404

**Pull request closed without merging**